tests/x11/firefox_audio.pm

```diff
@@ -18,16 +18,18 @@ use testapi;
 
 sub run {
     my ($self) = @_;
-    $self->start_firefox();
-    send_key "ctrl-l";
-    type_string(autoinst_url . "/data/1d5d9dD.oga");
-    send_key "ret";
     start_audiocapture;
+    x11_start_program("firefox " . autoinst_url . "/data/1d5d9dD.oga");
+    $self->firefox_check_default;
+    $self->firefox_check_popups;
     assert_screen 'test-firefox_audio-1', 35;
     sleep 1;    # at least a second of silence
     assert_recorded_sound('DTMF-159D');
     send_key "alt-f4";
-    $self->exit_firefox();
+    if (check_screen('firefox-save-and-quit', 4)) {
```

Comment:
please use multi-tag assert_screen with match_has_tag instead of check_screen with non-zero timeout.

Reply:
And once again, this won't be required if we will reuse exit_firefox method here.